practice-01/js/index.js

```diff
@@ -24,71 +29,162 @@ function showSuccess(input) {
+ */
+function displayEmptyError(resultCheckEmpty, listInput) {
+    listInput.forEach(input => {
+        input.value = input.value.trim();
+
+        if (resultCheckEmpty == false) {
+            showError(input, 'cannot be left blank')
+        } else {
+            showSuccess(input);
+        }
+    })
+}
+
 /**
- * Validation for Email
+ * Check the syntax mail which user fills in
+ * @param input : element of the argument
+ * @returns {boolean} Return true if text string match and pass the regex rule. Otherwise, return false.
  */
 function checkEmailError(input) {
+    //Regex : there must be a ‘@’ symbol after initial characters,email address must begin with alpha-numeric characters,it may have periods,underscores and hyphens.
     const regexEmail = /^\w+([\.-]?\w+)*@\w+([\.-]?\w+)*(\.\w{2,3})+$/;
     input.value = input.value.trim();
 
-    let isEmailError = !regexEmail.test(input.value);
-
     if (regexEmail.test(input.value)) {
-        isEmailError = true;
+        return true;
     } else {
-        isEmailError = false;
+        return false;
     }
-    return isEmailError;
 }
+
+/**
+ * Display an error message when the user enters the wrong email
+ * @param resultCheckEmail : true || false
+ * @param input : element of email ID
+ */
+function displayEmailError(resultCheckEmail, input) {
+    if (resultCheckEmail == false) {
+        showError(input, 'Email Invalid');
+    }
+}
+
 /**
- * Validation for charcaters
+ * Check special characters, text string match
+ * @param input : element of the argument
+ * @returns {boolean} Return true if text string match and pass the regex rule. Otherwise, return false.
  */
-function checkCharacter(input) {
-    const regexCharacter = /[ ^a-z-A-Z_ÀÁÂÃÈÉÊÌÍÒÓÔÕÙÚĂĐĨŨƠàáâãèéêìíòóôõùúăđĩũơƯĂẠẢẤẦẨẪẬẮẰẲẴẶẸẺẼỀỀỂưăạảấầẩẫậắằẳẵặẹẻẽềềểỄỆỈỊỌỎỐỒỔỖỘỚỜỞỠỢỤỦỨỪễếệỉịọỏốồổỗộớờởỡợụủứừỬỮỰỲỴÝỶỸửữựỳỵỷỹ]+$/;
+function checkNameError(input) {
+    // Regex : contains only letters, no numbers or special characters
+    const regexCharacter = /^[a-z ,.'-]+$/i;
 
     if (regexCharacter.test(input.value)) {
         return true;
     } else {
         return false;
     }
 }
+
 /**
- * Validation for limit of number
+ * Display a message to the user when the content contains special characters
+ * @param resultCheckName : true || false
+ * @param input : element of the argument
  */
-function checkLimitNumberError(input, min, max) {
+function displayNameError(resultCheckName, input) {
+    if (resultCheckName == false) {
+        showError(input, 'Name cannot contain characters');
+    }
+}
 
-    if (input.value < min) {
-        return false;
+/**
+ * Check if the value of the number meets the requirements or not
+ * @param value : element of the argument
+ * @param min : minimum value that the user passed in
+ * @param max : maximum value that the user passed in
+ * @returns {boolean} Returns true if the number is larger than min and less than maximum. Otherwise, return false.

```

Comment:
This one have `.` at the end of line. But the other dont have. Please make it consistent. Thanks